html/modules/custom/reliefweb_utility/src/Helpers/LocalizationHelper.php

```diff
@@ -134,42 +136,62 @@ public static function getCollator($language = NULL) {
     $language = static::getLanguage($language);
 
     if (!isset($collators[$language])) {
-      if (function_exists('collator_create')) {
-        $collator = collator_create($language);
-
-        switch (intl_get_error_code()) {
-          case U_ZERO_ERROR:
-            // No errors.
-            break;
-
-          case U_USING_DEFAULT_WARNING:
-            // For some reason, the French locale for the collation defaults
-            // to English and doesn't enable the French Collation. This is not
-            // an issue per se, as we can have the correct behavior by enabling
-            // the French collation so that accents are handled properly.
-            //
-            // @see https://www.php.net/manual/en/class.collator.php
-            if ($language === 'fr') {
-              $collator->setAttribute(Collator::FRENCH_COLLATION, Collator::ON);
-            }
-            break;
-
-          default:
-            // Some other error happened, we mark the collator as FALSE so that
-            // the collated_(k)sort functions can default to the basic (k)sort.
-            $collator = FALSE;
-        }
+      $collator = static::collatorCreate($language);
 
-        $collators[$language] = $collator;
+      if (!$collator) {
+        return FALSE;
```

Comment:
```suggestion
        $collators[$language] = FALSE;
        return FALSE;
```

In my opinion `$collators[$language]` should be set to `FALSE` in case of failure to avoid calling `collatorCreate` again.